core/database/foxx/api/record.js

```diff
@@ -0,0 +1,200 @@
+      ? basePath + this.#key
+      : basePath + "/" + this.#key;
+  }
+
+  /**
+   * @brief Compares two paths and if an error is detected will save the error code and message.
+   **/
+  _comparePaths(storedPath, inputPath) {
+    if (storedPath !== inputPath) {
+      this.#error = g_lib.ERR_PERM_DENIED;
+      this.#err_msg =
+        "Record path is not consistent with repo expected path is: " +
+        storedPath +
+        " attempted path is " +
+        inputPath;
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * @brief Checks if the record exists in the database.
+   * @return {boolean} True if the record exists, otherwise false.
+   */
+  exists() {
+    return this.#exists;
+  }
+
+  key() {
+    return this.#key;
+  }
+
+  id() {
+    return this.#data_id;
+  }
+  /**
+   * @brief Will return error code of last run method.
+   *
+   * If no error code, will return null
+   **/
+  error() {
+    return this.#error;
+  }
+
+  /**
+   * @brief Retrieves the error code of the last run method.
+   * @return {string|null} Error code or null if no error.
+   */
+  errorMessage() {
+    return this.#err_msg;
+  }
+
+  /**
+   * @brief Checks if the record is managed by DataFed.
+   * @return {boolean} True if managed, otherwise false.
+   */
+  isManaged() {
+    //{
+    //    _from: data._id,
+    //    _to: repo_alloc._to,
+    //    uid: owner_id
+    //};
+    this.#loc = g_db.loc.firstExample({
+      _from: this.#data_id,
+    });
+
+    if (!this.#loc) {
+      this.#error = g_lib.ERR_PERM_DENIED;
+      this.#err_msg =
+        "Permission denied data is not managed by DataFed. This can happen if you try to do a transfer directly from Globus.";
+      return false;
+    }
+    this.#alloc = g_db.alloc.firstExample({
+      _from: this.#loc.uid,
+      _to: this.#loc._to,
+    });
+
+    // If alloc is null then will return false if not null will return true.
+    return !!this.#alloc;
+  }
+
+  /**
+   * @brief Validates if the provided record path is consistent with the database.
+   * @param {string} a_path - The path to validate.
+   * @return {boolean} True if consistent, otherwise false.
+   */
+  isPathConsistent(a_path) {
+    // This function will populate the this.#loc member and the this.#alloc
+    // member
+    if (!this.isManaged()) {
+      return false;
+    }
+
+    // If path is missing the starting "/" add it back in
+    if (!a_path.startsWith("/") && this.#alloc.path.startsWith("/")) {
+      a_path = "/" + a_path;
+    }
+
+    // If there is a new repo we need to check the path there and use that
+    if (this.#loc.new_repo) {
+      // Below we get the allocation associated with data item by
+      // 1. Checking if the data item is in flight, is in the process
+      // of being moved to a new location or new owner and using that
+      // oweners id.
+      // 2. Using the loc.uid parameter if not inflight to get the owner
+      // id.
+      let new_alloc = g_db.alloc.firstExample({
```

Comment:
If we aren't going to reassign this, we can use `const` to assign the variable

```suggestion
      const new_alloc = g_db.alloc.firstExample({
```